File: src/users-java/src/main/java/com/users/application/handler/impl/UserHandler.java
package com.users.application.handler.impl;

import com.users.application.dto.request.UserRequestDto;
import com.users.application.dto.response.SuccessMessageResponseDto;
import com.users.application.dto.response.UserResponseDto;
import com.users.application.handler.IUserHandler;
import com.users.application.mapper.UserMapper;
import com.users.domain.api.IUserServicePort;
import com.users.domain.model.UserModel;

import java.time.LocalDateTime;

public class UserHandler implements IUserHandler {

    private IUserServicePort userServicePort;
    private UserMapper userMapper;

    public UserHandler(IUserServicePort userServicePort, UserMapper userMapper) {
        this.userServicePort = userServicePort;
        this.userMapper = userMapper;
    }

    @Override
    public SuccessMessageResponseDto deleteUser(String id) {
        userServicePort.deleteUser(id);
        return new SuccessMessageResponseDto("Usuario con Id: " +  id + " Eliminado", LocalDateTime.now());
    }

    @Override
    public SuccessMessageResponseDto updateUser(String id, UserRequestDto userRequestDto) {
        UserModel userModel = userMapper.toModel(userRequestDto);
        userServicePort.updateUser(id, userModel);
        return new SuccessMessageResponseDto("Usuario con el Id: " + id + " actualizado", LocalDateTime.now());
    }


}

File: src/users-node/application/handler/impl/UserHandler.ts
import { IUserServicePort } from "../../../domain/api/IUserServicePort";
import { UserRequestDto } from "../../dto/request/UserRequestDto";
import { UserResponseDto } from "../../dto/response/UserResponseDto";
import { UserMapper } from "../../mapper/UserMapper";
import { IUserHandler } from "../IUserHandler";

export class UserHandler implements IUserHandler {

    constructor(private userServicePort: IUserServicePort){}
    
    getUsers(): UserResponseDto[] {
        return this.userServicePort.getUsers().map(UserMapper.toResponse);
    }

    saveUser(userDto: UserRequestDto): UserResponseDto {
        const user = this.userServicePort.saveUser(UserMapper.toDomain(userDto))
        return UserMapper.toResponse(user);
    }
    
}